src/errors.rs

```diff
@@ -1,10 +1,21 @@
-use std::{cell::OnceCell, fmt, str::FromStr};
-
-use regex::Regex;
-
-use self::WhichLine::*;
+//! This module contains the logic for parsing rustc error messages.
+
+use {
+    self::WhichLine::*,
+    std::{fmt, str::FromStr},
+};
+
+// https://docs.rs/once_cell/1.19.0/once_cell/#lazily-compiled-regex
```

Comment:
```suggestion
/// Macro used to lazily create a new regex the first time it is invoked.
///
/// # Arguments
///
/// * `re` - The regex literal string used to build the automaton
///
/// # Example
///
/// ```rust
/// assert!(regex!(r"\w").is_match(" "));
/// ```
///
/// Taken from here https://docs.rs/once_cell/1.19.0/once_cell/#lazily-compiled-regex
```

Macros can be tricky to understand, I'd like this to be more thoroughly documented. A doctest example even provides a great insight (the example should compile & run when you launch cargo test).

Reply:
Sure thing, thanks for the brief example :)